students/hellosputnik/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+type Problem struct {
+	question string
+	answer   string
+}
+
+type Quiz struct {
+	problems []Problem
+	score    int
+}
+
+type Settings struct {
+	filename  *string
+	timeLimit *int
+}
+
+func main() {
+	quiz := Quiz{}
+	settings := Settings{}
+
+	settings.filename = flag.String("csv", "problems.csv", "a csv file in the format of 'question,answer'")
+	settings.timeLimit = flag.Int("limit", 30, "the time limit for the quiz in seconds")
```

Comment:
If you wanted you could use `flag.IntVar` and `flag.StringVar` here. Eg:

```go
type Settings struct {
	filename  string
	timeLimit int
}

func main() {
	var settings Settings
	flag.StringVar(&settings.filename, "csv", "problems.csv", "a csv file in the format of 'question,answer'")
	flag.IntVar(&settings.timeLimit, "limit", 30, "the time limit for the quiz in seconds")
	flag.Parse()
	fmt.Println(settings)
}
```

The primary upside is not needing to mess with string and integer pointers from here onwards for those values.